pkg/fleetautoscalers/fleetautoscalers_test.go

```diff
@@ -152,55 +79,592 @@ func (t testServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			f.Spec.Replicas = tc.specReplicas
+			f.Status.Replicas = tc.statusReplicas
+			f.Status.AllocatedReplicas = tc.statusAllocatedReplicas
+			f.Status.ReadyReplicas = tc.statusReadyReplicas
+
+			replicas, limited, err := applyWebhookPolicy(tc.webhookPolicy, f)
+
+			if tc.expected.err != "" && assert.NotNil(t, err) {
+				assert.Equal(t, tc.expected.err, err.Error())
+			} else {
+				assert.Equal(t, tc.expected.replicas, replicas)
+				assert.Equal(t, tc.expected.limited, limited)
+				assert.Nil(t, err)
+			}
+		})
+	}
+}
+
+func TestApplyWebhookPolicyNilFleet(t *testing.T) {
+	t.Parallel()
+
+	url := "/scale"
+	w := &autoscalingv1.WebhookPolicy{
+		Service: &admregv1b.ServiceReference{
+			Name:      "service1",
+			Namespace: "default",
+			Path:      &url,
+		},
+	}
+
+	replicas, limited, err := applyWebhookPolicy(w, nil)
+
+	if assert.NotNil(t, err) {
+		assert.Equal(t, "nil Fleet passed", err.Error())
+	}
+
+	assert.False(t, limited)
+	assert.Zero(t, replicas)
+}
+
+func TestCreateURL(t *testing.T) {
+	t.Parallel()
+
+	var testCases = []struct {
+		description string
+		scheme      string
+		name        string
+		namespace   string
+		path        string
+		expected    string
+	}{
+		{
+			description: "OK, path not empty",
+			scheme:      "http",
+			name:        "service1",
+			namespace:   "default",
+			path:        "scale",
+			expected:    "http://service1.default.svc:8000/scale",
+		},
+		{
+			description: "OK, path not empty with slash",
+			scheme:      "http",
+			name:        "service1",
+			namespace:   "default",
+			path:        "/scale",
+			expected:    "http://service1.default.svc:8000/scale",
+		},
+		{
+			description: "OK, path is empty",
+			scheme:      "http",
+			name:        "service1",
+			namespace:   "default",
+			path:        "",
+			expected:    "http://service1.default.svc:8000",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			res := createURL(tc.scheme, tc.name, tc.namespace, tc.path)
+
+			if assert.NotNil(t, res) {
+				assert.Equal(t, tc.expected, res.String())
+			}
+		})
+	}
+}
+
+func TestBuildURLFromWebhookPolicyNoNamespace(t *testing.T) {
+	url := "/testurl"
+
+	type expected struct {
+		url string
+		err string
+	}
+
+	var testCases = []struct {
+		description   string
+		webhookPolicy *autoscalingv1.WebhookPolicy
+		expected      expected
+	}{
+		{
+			description: "No namespace provided, default should be used",
+			webhookPolicy: &autoscalingv1.WebhookPolicy{
+				Service: &admregv1b.ServiceReference{
+					Name:      "service1",
+					Namespace: "",
+					Path:      &url,
+				},
+			},
+			expected: expected{
+				url: "http://service1.default.svc:8000/testurl",
+				err: "",
+			},
+		},
+		{
+			description: "No url provided, empty string should be used",
+			webhookPolicy: &autoscalingv1.WebhookPolicy{
+				Service: &admregv1b.ServiceReference{
```

Comment:
Same as above: Please add one more test case here, when both Service and URL would be nil

Reply:
Actually it's enough to add this test in one test suit.